### PR TITLE
User registartion workflow

### PR DESCRIPTION
**Pull request by rohitbhalke**

### CODE REVIEW

File: src/main/java/org/onedatashare/server/model/util/Response.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package org.onedatashare.server.model.util;
+
+
```
Comment (rohitbhalke) [inline src/main/java/org/onedatashare/server/model/util/Response.java:3]: Created a new class in order to have custom Response object

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 2]
```diff
@@ -45,13 +51,48 @@ public UserService(UserRepository userRepository) {
...
+    /*
+      This would be a same temporary password for each user while creating,
+      once the user goes through the whole User creation workflow, he/she can change the password.
+     */
+      User user = new User(email, password);
+      String token = user.validationToken();
+      user.setValidationToken(token);
+      String code = RandomStringUtils.randomAlphanumeric(6);
+      user.setVerifyCode(code);
+      user.registerMoment = new Date().getTime();
```
Comment (rohitbhalke) [inline src/main/java/org/onedatashare/server/service/UserService.java:73]: registerMoment is the field on user object. First we create a user record in db with current timestamp. Once user verifies his account and sends his own password, we change registerMoment field to 0. 
This helps in identifying records which are not verified.

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 3]
```diff
@@ -45,13 +51,48 @@ public UserService(UserRepository userRepository) {
...
+  public Mono<Boolean> setPassword(String email, String password, String passwordConfirm){
+    return getUser(email).flatMap(user-> {
+      if(!password.equals(passwordConfirm)){
+        return Mono.error(new Exception("Password is not confirmed."));
+      }else if(user.getAuthToken() == null){
+        return Mono.error(new Exception("Does not have Auth Token"));
+      }else {
+        user.setPassword(password);
+        user.setAuthToken(null);
+        user.registerMoment = 0;
```
Comment (rohitbhalke) [inline src/main/java/org/onedatashare/server/service/UserService.java:88]: Setting registerMoment to 0, once user verifies his/her account.

File: src/main/java/org/onedatashare/server/service/UserService.java [hunk 4]
```diff
@@ -45,13 +51,48 @@ public UserService(UserRepository userRepository) {
...
+      once the user goes through the whole User creation workflow, he/she can change the password.
+     */
+      User user = new User(email, password);
+      String token = user.validationToken();
+      user.setValidationToken(token);
+      String code = RandomStringUtils.randomAlphanumeric(6);
+      user.setVerifyCode(code);
+      user.registerMoment = new Date().getTime();
+      return createUser(user).flatMap(createdUser-> {
+        System.out.println("IN");
```
Comment (LinusCastelino) [inline src/main/java/org/onedatashare/server/service/UserService.java]: Is this print statement needed? Can we take it off?

Comment (rohitbhalke) [reply src/main/java/org/onedatashare/server/service/UserService.java]: Yes. Sorry, i am removing it

**Review by yinyifu — APPROVED** [review]
Hope sendVerificationCode1 method names can be named more meaningful.

**Comment by yinyifu** [comment]
Fixed
#25

**Pull request merged**